semana10/projeto-labex/labex/src/assets/components/CreateTripPage.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import axios from "axios";
+import { useHistory } from "react-router-dom";
+import { useProtectPage } from "../hooks/useProtectPage"
+import {baseUrl} from '../constants/axiosConstants'
+import TextField from '@material-ui/core/TextField';
+import Select from '@material-ui/core/Select'
+import Button from '@material-ui/core/Button';
+import { styled } from '@material-ui/styles'
+import Grid from '@material-ui/core/Grid'
+import useForm from '../hooks/useForm'
+import { useState, useEffect } from 'react'
+import { InputLabel, FormControl } from '@material-ui/core';
+
+const FormDiv = styled(Grid)({
+    display: "grid",
+    gap: "1em",
+    alignItems: "center",
+    justifyItems: "center",
+    width: "100vw",
+    marginBottom: "50px"
+})
+
+const InputField = styled(TextField)({
+    width: "30vw",
+})
+
+const NamesInputField = styled(TextField)({
+    width: "30vw",
+})
+
+const DescriptionInputField = styled(TextField)({
+    width: "30vw",
+})
+const SelectOption = styled(Select)({
+  width: "30vw",
+  backgroundColor: "#FFFFFF",
+})
+
+
+export default function CreateTripPage() {
+    const history = useHistory();
+
+    useEffect(() => {
+        const token = window.localStorage.getItem("token");
+    
+        if (token) {
+         
+        } else {
+          history.push("/login");
+        }
```

Comment:
Aproveita a oportunidade pra usar um curto circuito, fica bem simples.
`!token && history.push("/login")`

Reply:
não me acostumei muito com curto circuito e ternário ainda, to engatinhando